src/server/json_family.cc

```diff
@@ -554,32 +476,34 @@ size_t CountJsonFields(const JsonType& j) {
   return res;
 }
 
-void SendJsonValue(RedisReplyBuilder* rb, const JsonType& j) {
-  if (j.is_double()) {
-    rb->SendDouble(j.as_double());
-  } else if (j.is_number()) {
-    rb->SendLong(j.as_integer<long>());
-  } else if (j.is_bool()) {
-    rb->SendSimpleString(j.as_bool() ? "true" : "false");
-  } else if (j.is_null()) {
-    rb->SendNull();
-  } else if (j.is_string()) {
-    rb->SendBulkString(j.as_string_view());
-  } else if (j.is_object()) {
-    rb->StartArray(j.size() + 1);
-    rb->SendSimpleString("{");
-    for (const auto& item : j.object_range()) {
-      rb->StartArray(2);
-      rb->SendBulkString(item.key());
-      SendJsonValue(rb, item.value());
+template <typename T, bool ReturnEmptyResultIfKeyNotFound = false,
+          bool ReturnWrongTypeOnNullOpt = false>
+OpResult<JsonCallbackResult<T>> JsonEvaluateOperation(const OpArgs& op_args, std::string_view key,
+                                                      const WrappedJsonPath& json_path,
+                                                      JsonPathEvaluateCallback<T> cb,
+                                                      bool save_first_result = false) {
```

Comment:
A bit unusual to see control flow like this with non type templates 🙂

Reply:
just use an `enum class` and pass it as a `parameter`. Making this a template is an overkill

Reply:
The problem was that `ReturnWronTypeOnNullopt` can only be compiled when `T` =`std::optional<SomeType>`. But I changed it to a different way.